flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -540,8 +540,8 @@ public final void invoke() throws Exception {
 			try {
 				cleanUpInvoke();
 			}
-			catch (Throwable cleanUpException) {
-				throw (Exception) ExceptionUtils.firstOrSuppressed(cleanUpException, invokeException);
+			catch (Exception cleanUpException) {
+				throw ExceptionUtils.firstOrSuppressed(cleanUpException, invokeException);
```

Comment:
I think we agreed during the offline discussion to preserve the types of what is being caught (or am I wrong?).

We can preserve `Throwable` like this:
```
} catch (Throwable cleanUpException) {
    ExceptionUtils.rethrow(ExceptionUtils.firstOrSuppressed(cleanUpException, invokeException));
```

I think this should be a separate `[hotfix]` commit.